src/code_fa4c0_len_3bf0.c

```diff
@@ -22,23 +22,60 @@ ApiStatus SetMusicTrack(ScriptInstance* script, s32 isInitialCall) {
     return (set_music_track(musicPlayer, songID, variation, 0x1F4, volume) != 0) * 2;
 }
 
-INCLUDE_ASM(s32, "code_fa4c0_len_3bf0", FadeInMusic, ScriptInstance* script, s32 isInitialCall);
+ApiStatus FadeInMusic(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 var0 = get_variable(script, *args++);
+    s32 var1 = get_variable(script, *args++);
+    s32 var2 = get_variable(script, *args++);
+    s32 var3 = get_variable(script, *args++);
+    s16 var4 = get_variable(script, *args++);
+    s16 var5 = get_variable(script, *args++);
+
+    return (func_8014A964(var0, var1, var2, var3, var4, var5) != 0) * 2;
+}
 
 INCLUDE_ASM(s32, "code_fa4c0_len_3bf0", func_802D5EE0);
 
-INCLUDE_ASM(s32, "code_fa4c0_len_3bf0", func_802D5F28);
+ApiStatus func_802D5F28(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+
+    func_8014AA54(get_variable(script, *args++), get_variable(script, *args++), (s16)get_variable(script, *args++));
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802D5FA4(ScriptInstance* script, s32 isInitialCall) {
+    func_8014AB0C(0, (s16)get_variable(script, *script->ptrReadPos));
```

Comment:
Probably worth typing this function so the explicit cast isn't required. Same with func_8014AA54 above.

Reply:
I would normally agree, but given this is our only call to this function, it's not clear yet if it takes an s16 or it doesn't and the cast is actually not needed. I'd rather not assume yet